expertisesolutions/efl
# Windows:eina: Add Thread synchronization

Summary:
========
This should add all needed functions/structures for thread synchronization besides:
- eina_condition initialization;
- cond_timedwait;

Windows Test Plan:
==================
- comment `Wno-implicit-function-declaration` from meson.build;
- build.bat with -k0;
- there should not be any errors/warnings about any synchronization or
  TLS related function;
- It should compile the same amount of files than
  `devs/expertise/native-windows`;

Linux Test Plan:
======================
- rebase #39 
- configure meson with `bindigns=cxx,mono`;
- `ninja` should return only erros/wargings that `master` does;
- `meson test all` should only pass on tests that master does;
